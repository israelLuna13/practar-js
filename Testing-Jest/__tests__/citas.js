import Citas from "../js/classes/Citas";
const id = Date.now()

describe("Probar la clase de citas",()=>{

    const citas = new Citas()
    test('Agregar una nueva cita', ()=>{

    const citaObj = {
        id,
        mascota: 'hook',
        propietario: 'israel',
        telefono: '22222222',
        fecha: '10-20-2026',
        hora:'10:20',
        sintomas: 'jajajajjaja'
    }
    citas.agregarCita(citaObj)

    expect(citas).toMatchSnapshot()
    })

    test("actualizar cita", ()=>{
        const citaActualizada = {
            id:id,
            mascota: 'escubi',
            propietario: 'israel',
            telefono: '22222222',
            fecha: '10-20-2026',
            hora:'10:20',
            sintomas: 'jajajajjaja'
        }

        citas.editarCita(citaActualizada)
        expect(citas).toMatchSnapshot()
    })

    test("eliminar cita", ()=>{
      citas.eliminarCita(id)
      expect(citas).toMatchSnapshot()
    })
})