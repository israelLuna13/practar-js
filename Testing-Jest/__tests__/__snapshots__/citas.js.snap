// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-20-2026",
      "hora": "10:20",
      "id": 1739666873366,
      "mascota": "hook",
      "propietario": "israel",
      "sintomas": "jajajajjaja",
      "telefono": "22222222",
    },
  ],
}
`;

exports[`Probar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-20-2026",
      "hora": "10:20",
      "id": 1739666873366,
      "mascota": "escubi",
      "propietario": "israel",
      "sintomas": "jajajajjaja",
      "telefono": "22222222",
    },
  ],
}
`;

exports[`Probar la clase de citas eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
